src/TimeoutError.ts

```diff
@@ -2,9 +2,9 @@ export default class TimeoutError extends Error {
   public timeout: number;
   public statusCode: number;
 
-  constructor(message: string, timeout?: number, statusCode?: number) {
```

Comment:
I'm surprised the optional `timeout` param wasn't causing a TS error here 🤔 

I thought the potentially `undefined` timeout would conflict with `public timeout: number;`

Reply:
That's a good point. Since the parameter was optional, it could be undefined but there'd probably be a runtime error when the value is assigned to `this.timeout`. I'm not sure why either of those were flagged as optional. I think that timeout should be required and statusCode should have a default value.